File: Interpreter/c#/Interpreter.cs
interface IExpression
{
    public void Interpret();
}

class TerminalExpression : IExpression
{
    private string _statement;
    
    public TerminalExpression(string statement) => _statement = statement;

    public void Interpret()
    {
        Console.WriteLine(this._statement + " TerminalExpression");
    }
}

class NonTerminalExpression : IExpression
{
    private IExpression _expression;
    private int _times;

    public NonTerminalExpression(IExpression expression, int times)
    {
        _expression = expression;
        _times = times;
    }

    public void Interpret()
    {
        for (int i = 0; i < _times; i++)
        {
            this._expression.Interpret();
        }
    }
}

class Program
{
    static void Main()
    {
        IExpression terminalExpression = new TerminalExpression("hello");
        IExpression nonTerminalExpression = new NonTerminalExpression(terminalExpression, 2);
        
        nonTerminalExpression.Interpret();
    }
}

File: Interpreter/java/Interpreter.java
interface IExpression{
    public void interpret();
}

class TerminalExpression implements IExpression{
    private String statement;

    public TerminalExpression(final String statement){
        this.statement = statement;
    }

    @Override
    public void interpret(){
        System.out.println(this.statement + " TerminalExpression");
    }
}

class NonTerminalExpression implements IExpression{
    private IExpression expression = null;
    private int times;

    public NonTerminalExpression(IExpression expression, int times){
        this.expression = expression;
        this.times = times;
    }
    
    @Override
    public void interpret(){
        for (int i = 0; i < this.times; i++){
            this.expression.interpret();
        }
    }
}

public class Main {
    public static void main(String[] args){
        IExpression terminalExpression = new TerminalExpression("hello");
        IExpression nonTerminalExpression = new NonTerminalExpression(terminalExpression, 2);

        nonTerminalExpression.interpret();
    }
}